lib/internal/modules/esm/loader.js

```diff
@@ -215,7 +220,14 @@ class Loader {
     if (!this._getGlobalPreloadCode) {
       return;
     }
-    const preloadCode = this._getGlobalPreloadCode();
+    const channel = new MessageChannel();
+    const insidePreload = channel.port1;
+    insidePreload.unref();
+    const insideLoader = channel.port2;
+    insideLoader.unref();
+    const preloadCode = this._getGlobalPreloadCode({
+      port: insideLoader
+    });
```

Comment:
this is gonna conflict with #37468 which renames this method

Reply:
I'm fine rebasing this after that PR lands. I suspect the timing will make this be a non issue but we could simply mark it as blocking. Either way, one of these 2 PRs needs to be rebased since they won't land at the same time.

Reply:
rebased